tools/explorer/lib/ontology_wrapper.py

```diff
@@ -105,58 +103,225 @@ def GetFieldsForTypeName(
+        concrete_fields.difference(standard_canonical_fields)
+    )
+    unmatched_required_fields = len(
+        required_canonical_fields.difference(concrete_fields)
+    )
+
+    if total_entity_fields <= 0:
+      raise ValueError('Concrete field set cannot be empty.')
+
+    total_precision = matched_fields - unmatched_entity_fields
+    total_precision /= total_entity_fields
+
+    if total_required_type_fields <= 0:
+      match_score = total_precision / 2.0
+    else:
+      required_precision = matched_required_fields - unmatched_required_fields
+      required_precision /= total_required_type_fields
+      match_score = (total_precision + required_precision) / 2.0
+    return int((match_score + 1.0) * 50)
+
+  def _CreateMatch(self, field_list: List[StandardField],
                    entity_type: EntityType) -> Match:
-    """Determines the match between: EntityType and EntityTypeFields.
+    """Creates a Match instance for an EntityType object and a list of
+    StandardField objects.
 
     calls _CalculateMatchWeight() on field_list and the set of fields belonging
-    to entity_type. The weight function outputs a weight signifying the
-    closeness of the match and an instance of the Match class is created with
-    field_list, entity_type, and weight as arguments.
+    to entity_type. The scoring function outputs a float signifying the
+    closeness of the match, and an instance of the Match class is created with
+    field_list, entity_type, and match_score as arguments.
 
     Args:
-      field_list: A list of EntityTypeField objects.
+      field_list: A list of EntityTypeField objects for a concrete entity.
       entity_type: An EntityType object.
 
     Returns:
-      An instance of the Match class.
+      An instance of Match class.
     """
-    pass
+    canonical_field_set = set()
+    for qualified_field in entity_type.GetAllFields().values():
+      new_entity_type_field = EntityTypeField(
+          qualified_field.field.namespace,
+          qualified_field.field.field,
+          qualified_field.optional,
+          qualified_field.field.increment
+      )
+      canonical_field_set.add(new_entity_type_field)
+
+    match_score = self._CalculateMatchScore(
+        concrete_fields=frozenset(field_list),
+        canonical_fields=frozenset(canonical_field_set)
+    )
+    new_match = Match(
+        field_list,
+        entity_type,
+        match_score
+    )
+    return new_match
 
   def GetEntityTypesFromFields(self,
-                               field_list: List[EntityTypeField],
-                               general_type: str = None) -> List[EntityType]:
-    """Get a list of EntityType objects matching a list of EntityTypeFields.
+                               field_list: List[StandardField],
+                               return_size: int = 0,
+                               general_type: str = None) -> List[Match]:
+    """Get a list of Match objects containg information on a strength of a
```

Comment:
what is a strength? in
    [0, 100].

Reply:
is it more of a score?

Reply:
I think it might be easier here to be less verbose. If the user is curious as to what a match is, they can look at model.Match